android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt

```diff
@@ -282,21 +276,8 @@ class RNOneginiSdk(private val reactContext: ReactApplicationContext) : ReactCon
     @ReactMethod
     fun startSingleSignOn(uri: String?, promise: Promise) {
         when (uri) {
-            null -> promise.rejectWithNullError(FunctionParams.Uri.paramName, FunctionParams.Uri.type)
-            else -> {
-                oneginiSDK.oneginiClient.userClient.getAppToWebSingleSignOn(
-                    Uri.parse(uri),
-                    object : OneginiAppToWebSingleSignOnHandler {
-                        override fun onSuccess(oneginiAppToWebSingleSignOn: OneginiAppToWebSingleSignOn) {
-                            promise.resolve(OneginiAppToWebSingleSignOnMapper.toWritableMap(oneginiAppToWebSingleSignOn))
-                        }
-
-                        override fun onError(error: OneginiAppToWebSingleSignOnError) {
-                            promise.reject(error.errorType.toString(), error.message)
-                        }
-                    }
-                )
-            }
+            null -> promise.rejectWithNullError(Uri.paramName, Uri.type)
+            else -> sdkWrapper.startSingleSignOn(uri, promise)
```

Comment:
with logic extracted into separate class, I think we could go with classic elvis operator here
```
uri?.let { sdkWrapper.startSingleSignOn(uri, promise) } ?: promise.rejectWithNullError(Uri.paramName, Uri.type)
```